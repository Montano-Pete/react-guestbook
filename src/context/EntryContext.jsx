import { createContext, useState } from 'react';

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
  const [entry, setEntry] = useState();

  return (
    <EntryContext.Provider value={{ entry, setEntry }}>
      {children}
    </EntryContext.Provider>
  );
};

export { EntryProvider, EntryContext };
