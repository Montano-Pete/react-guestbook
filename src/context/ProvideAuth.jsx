import React from 'react';
import { createContext, useState } from 'react';

const authContext = createContext();

function ProvideAuth({ children }) {
  const [user, setUser] = useState(null);

  const login = (email, password) => {
    const loginSuccessful =
      email === process.env.AUTH_EMAIL &&
      password === process.env.AUTH_PASSWORD;
    if (loginSuccessful) setUser({ email });
    return loginSuccessful;
  };

  const logout = (callback) => {
    setUser(null);
    callback();
  };

  return (
    <authContext.Provider value={{ user, login, logout }}>
      {children}
    </authContext.Provider>
  );
}

export { authContext, ProvideAuth };
