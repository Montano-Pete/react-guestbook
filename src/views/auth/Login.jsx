import React, { useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import useAuth from '../../hooks/useAuth';
import useForm from '../../hooks/useForm';
import useUser from '../../hooks/useUser';

function Login() {
  const history = useHistory();
  const location = useLocation();
  const auth = useAuth();

  const { formState, handleFormChange } = useForm({ email: '', password: '' });
  const [error, setError] = useState(null);
  const { from } = location.state || { from: { pathname: '/' } };
  const { setUser } = useUser();

  const handleLogin = (event) => {
    event.preventDefault();
    const loginWasSuccessful = auth.login(formState.email, formState.password);

    if (loginWasSuccessful) {
      setUser(formState.email);
      history.replace(from.pathname);
    } else {
      setError(
        'Incorrect login information. Please double-check and try again!'
      );
    }
  };

  return (
    <>
      <fieldset>
        <legend>Login here to sign!</legend>
        <p>{error}</p>
        <form onSubmit={handleLogin}>
          <label htmlFor="email">Email</label>
          <input
            id="email"
            name="email"
            type="email"
            value={formState.email}
            onChange={handleFormChange}
            placeholder="Enter email"
            required
          />
          <label htmlFor="password">Password</label>
          <input
            id="password"
            name="password"
            type="password"
            value={formState.password}
            onChange={handleFormChange}
            placeholder="Enter Password"
            required
          />
          <button type="submit">Sign In</button>
        </form>
      </fieldset>
    </>
  );
}

export default Login;
