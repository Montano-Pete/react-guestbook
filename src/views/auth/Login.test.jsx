import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { ProvideAuth } from '../../context/ProvideAuth';
import { UserProvider } from '../../context/UserContext';
import Login from './Login';

it('should render the login page', () => {
  const container = render(
    <MemoryRouter>
      <UserProvider>
        <ProvideAuth>
          <Login />
        </ProvideAuth>
      </UserProvider>
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
