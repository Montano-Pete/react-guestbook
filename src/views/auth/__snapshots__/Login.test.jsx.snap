// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset>
        <legend>
          Login here to sign!
        </legend>
        <p />
        <form>
          <label
            for="email"
          >
            Email
          </label>
          <input
            id="email"
            name="email"
            placeholder="Enter email"
            required=""
            type="email"
            value=""
          />
          <label
            for="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
            placeholder="Enter Password"
            required=""
            type="password"
            value=""
          />
          <button
            type="submit"
          >
            Sign In
          </button>
        </form>
      </fieldset>
    </div>
  </body>,
  "container": <div>
    <fieldset>
      <legend>
        Login here to sign!
      </legend>
      <p />
      <form>
        <label
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          name="email"
          placeholder="Enter email"
          required=""
          type="email"
          value=""
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          placeholder="Enter Password"
          required=""
          type="password"
          value=""
        />
        <button
          type="submit"
        >
          Sign In
        </button>
      </form>
    </fieldset>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
